### PR TITLE
Handle flawfinder severity level

### PR DESCRIPTION
**Pull request by Christian-Gibbons**
This should allow a user to specify a severity level to be considered an error while lesser issues would be treated as warnings.  It also adds a little more information to issue.  Intended to solve issue #1385 

Not sure if I handled the Vader test properly for this one.  Most of the changes dealt with the GCC handler rather than the flawfinder linter itself.

### CODE REVIEW

File: ale_linters/c/flawfinder.vim [hunk 1]
```diff
@@ -4,6 +4,7 @@
 call ale#Set('c_flawfinder_executable', 'flawfinder')
 call ale#Set('c_flawfinder_options', '')
 call ale#Set('c_flawfinder_minlevel', 1)
+call ale#Set('flawfinder_error_severity', 6)
```
Comment (w0rp) [inline ale_linters/c/flawfinder.vim]: This should be named `c_flawfinder_error_severity`.

Comment (Christian-Gibbons) [reply ale_linters/c/flawfinder.vim]: I was unsure about this one.  This variable affects both C and C++.

File: autoload/ale/handlers/gcc.vim [hunk 2]
```diff
@@ -24,7 +24,7 @@ function! ale#handlers#gcc#HandleGCCFormat(buffer, lines) abort
     " <stdin>:8:5: warning: conversion lacks type at end of format [-Wformat=]
     " <stdin>:10:27: error: invalid operands to binary - (have ‘int’ and ‘char *’)
     " -:189:7: note: $/${} is unnecessary on arithmetic variables. [SC2004]
-    let l:pattern = '\v^([a-zA-Z]?:?[^:]+):(\d+):(\d+)?:? ([^:]+): ?(.+)$'
+    let l:pattern = '\v^([a-zA-Z]?:?[^:]+):(\d+):(\d+)?:? ([^:]+): (.+)$'
```
Comment (w0rp) [inline autoload/ale/handlers/gcc.vim]: Don't forget to undo the changes to this file too.

Comment (Christian-Gibbons) [reply autoload/ale/handlers/gcc.vim]: That change was intentional.  I added the question mark to that regular expression when I added the initial Flawfinder support that piggybacked off of the GCC handler in #1361 .   Now that Flawfinder has its own handler, I figured I would revert that back to what it had been before I commandeered the GCC handler for Flawfinder.  I can add it back in if you like, but there shouldn't be anything relying on that question mark anymore.

Comment (w0rp) [reply autoload/ale/handlers/gcc.vim]: Okay, what about the other changes below?

Comment (Christian-Gibbons) [reply autoload/ale/handlers/gcc.vim]: Hmm, not sure how I missed all that.  I'll try to get a fix in later today.
